File: more-zeroes/__tests__/moreZeroes.test.js
const { moreZeroes } = require('../moreZeroes');

describe(`Create a moreZeros function which will receive a string for input, and return an array (or null terminated string in C) containing only the characters from that string whose binary representation of its ASCII value consists of more zeros than ones.
        You should remove any duplicate characters, keeping the first occurrence of any such duplicates, so they are in the same order in the final array as they first appeared in the input string.`, () => {
    test("Given 'abcde', return ['a','b','d']", () => {
        const string = "abcde";
        expect(moreZeroes(string)).toEqual(['a', 'b', 'd']);
    });
    test("Given 'Great job!'), return ['a', ' ', 'b', '!']", () => {
        const string = "Great job!";
        expect(moreZeroes(string)).toEqual(['a', ' ', 'b', '!']);
    });
    test("Given 'DIGEST'), return ['D','I','E','T']", () => {
        const string = "DIGEST";
        expect(moreZeroes(string)).toEqual(['D', 'I', 'E', 'T']);
    });
    test("Given 'abcdeabcde'), return ['a','b','d'],'Should not return duplicates values'", () => {
        const string = "abcdeabcde";
        expect(moreZeroes(string)).toEqual(['a', 'b', 'd']);
    });
    test("Given 'Forgiveness is the fragrance that the violet sheds on the heal that has crushed it', return ['F', ' ', 'h', 'a', 'd']", () => {
        const string = "Forgiveness is the fragrance that the violet sheds on the heal that has crushed it";
        expect(moreZeroes(string)).toEqual(["F", " ", "h", "a", "d"]);
    });
    test("Given 'abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ1234567890_', return ['a', 'b', 'd', 'h', 'p', 'A', 'B', 'C', 'D', 'E', 'F', 'H', 'I', 'J', 'L', 'P', 'Q', 'R', 'T', 'X', '0']", () => {
        const string = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ1234567890_";
        expect(moreZeroes(string)).toEqual(['a', 'b', 'd', 'h', 'p', 'A', 'B', 'C', 'D', 'E', 'F', 'H', 'I', 'J', 'L', 'P', 'Q', 'R', 'T', 'X', '0']);
    });
    test("Given 'thequickbrownfoxjumpsoverthelazydog', return ['h', 'b', 'p', 'a', 'd']", () => {
        const string = "thequickbrownfoxjumpsoverthelazydog";
        expect(moreZeroes(string)).toEqual(['h', 'b', 'p', 'a', 'd']);
    });
    test(`Given "THEQUICKBROWNFOXJUMPSOVERTHELAZYDOG", return ['T', 'H', 'E', 'Q', 'I', 'C', 'B', 'R', 'F', 'X', 'J', 'P', 'L', 'A', 'D']`, () => {
        const string = "THEQUICKBROWNFOXJUMPSOVERTHELAZYDOG";
        expect(moreZeroes(string)).toEqual( ['T', 'H', 'E', 'Q', 'I', 'C', 'B', 'R', 'F', 'X', 'J', 'P', 'L', 'A', 'D']);
    });
});

//https://www.codewars.com/kata/5d41e16d8bad42002208fe1a/train/javascript

File: python-venv/more-zeroes/more_zeroes_test.py
from more_zeroes import more_zeroes

def test_1():
    string = 'abcde'
    assert more_zeroes(string) == ['a', 'b', 'd']

def test_2():
    string = 'Great job!'
    assert more_zeroes(string) == ['a', ' ', 'b', '!']

def test_3():
    string = 'DIGEST'
    assert more_zeroes(string) == ['D', 'I', 'E', 'T']

def test_4():
    string = 'abcdeabcde'
    assert more_zeroes(string) == ['a', 'b', 'd']

def test_5():
    string = 'Forgiveness is the fragrance that the violet sheds on the heal that has crushed it'
    assert more_zeroes(string) == ["F", " ", "h", "a", "d"]

def test_6():
    string = 'abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ1234567890_'
    assert more_zeroes(string) == ['a', 'b', 'd', 'h', 'p', 'A', 'B', 'C', 'D', 'E', 'F', 'H', 'I', 'J', 'L', 'P', 'Q', 'R', 'T', 'X', '0']

def test_7():
    string = 'thequickbrownfoxjumpsoverthelazydog'
    assert more_zeroes(string) == ['h', 'b', 'p', 'a', 'd']

def test_8():
    string = 'THEQUICKBROWNFOXJUMPSOVERTHELAZYDOG'
    assert more_zeroes(string) == ['T', 'H', 'E', 'Q', 'I', 'C', 'B', 'R', 'F', 'X', 'J', 'P', 'L', 'A', 'D']


#https://www.codewars.com/kata/5d41e16d8bad42002208fe1a/train/python